src/main/scala/au/com/agiledigital/toolform/app/ToolFormApp.scala

```diff
@@ -1,110 +1,32 @@
-    }
-
-    parser.parse(args, ToolFormConfiguration()).toRight(ToolFormError(s"Invalid arguments - toolform failed."))
-  }
-
-  private def readProject(projectDefinitionFile: File): Either[ToolFormError, Project] =
-    if (!projectDefinitionFile.exists()) {
-      Left(ToolFormError(s"File [$projectDefinitionFile] does not exist."))
-    } else if (!projectDefinitionFile.isFile) {
-      Left(ToolFormError(s"File [$projectDefinitionFile] is not a file."))
-    } else {
-      // Read in the root configuration file.
-      Try(ConfigFactory.parseFile(projectDefinitionFile)) match {
-        case Success(simpleConfig) =>
-          // Resolve the configuration aka. replace variable substitutions.
-          val config = simpleConfig.resolve()
-
-          val projectResult = loadConfig[Project](config)
-          resultOrCollectReadErrors(projectResult)
-        case Failure(e) =>
-          Left(ToolFormError(s"Failed to parse project configuration [$projectDefinitionFile]: [${e.getMessage}]"))
-      }
-    }
-
-  private def resultOrCollectReadErrors(projectResult: Either[ConfigReaderFailures, Project]): Either[ToolFormError, Project] =
-    projectResult.left.map(failures => {
-      val failureDetails: String = failures.toList.map(failure => {
-        val locationDescription = failure.location.map(_.description).getOrElse("Unknown location")
-        val failureMessage: String = failure.description + " @ " + locationDescription
-        failure match {
-          case KeyNotFound(key, _, _) => s"[$key] $failureMessage"
-          case _                      => failureMessage
-        }
-      }).mkString(":\n")
-      ToolFormError(s"Failed to read project: [$failureDetails]")
-    })
-
-  private def projectSummary(project: Project): Either[ToolFormError, String] = {
-    val projectComponentsSummary = project.components.values.map(c => s"${c.id} ==> '${c.name}'").mkString("\n\t\t")
-    val projectResourcesSummary = project.resources.values.map(r => r.id).mkString("\n\t\t")
-    val projectLinksSummary = project.topology.links.map(l => {
-      val resolvedLink = l.resolve(project)
-      s"${resolvedLink.from.id} -> ${resolvedLink.to.id}"
-    }).mkString("\n\t\t")
-    Right(
-      s"Project: [${project.name}]\n" +
-      s"\tComponents:\n\t\t$projectComponentsSummary\n" +
-      s"\tResources:\n\t\t$projectResourcesSummary\n" +
-      s"\tLinks:\n\t\t$projectLinksSummary\n")
-  }
-
-  execute(args) match {
-    case Left(error) =>
-      Console.err.println(error.message)
-      sys.exit(1)
-    case Right(result) =>
-      println(result)
-  }
+object ToolFormApp
+    extends CommandApp(
+      name = BuildInfo.name,
+      version = BuildInfo.version,
+      header = "Generates deployment configuration from a project definition.",
+      main = ToolFormPluginLoader.loadPlugins.map(plugin => plugin.command).foldLeft(Opts.unit)(_ orElse _)
+    )
+
+object ToolFormPluginLoader {
+  def loadPlugins: List[ToolFormPlugin] =
```

Comment:
The immutable `Seq` is probably better here. Take a look at https://docs.scala-lang.org/overviews/collections/overview.html to see how Java collection types map into scala.